x-pack/plugins/alerting/server/task_runner/alert_task_instance.ts

```diff
@@ -42,15 +36,6 @@ export function taskInstanceToAlertTaskInstance<Params extends RuleTypeParams>(
         );
       }, t.identity)
     ),
-    state: pipe(
-      ruleStateSchema.decode(taskInstance.state),
-      fold((e: t.Errors) => {
-        throw new Error(
-          `Task "${taskInstance.id}" ${
-            alert ? `(underlying Alert "${alert.id}") ` : ''
-          }has invalid state at ${enumerateErrorFields(e)}`
-        );
-      }, t.identity)
-    ),
```

Comment:
Logic used to convert date strings into dates. We now type them as strings all over, so this is no longer needed.